storage/index.go

```diff
@@ -46,50 +60,120 @@ const (
 	nsIndexStateClosed
 )
 
+// NB: `latestBlock` v `blocksByTime`: blocksByTime contains all the blocks known to `nsIndex`.
+// `latestBlock` refers to the block with greatest StartTime within blocksByTime. We do this
+// to skip accessing the map blocksByTime in the vast majority of write/query requests. It's
+// lazily updated, so it can point to an older element until a Tick()/write rotates it.
+// nolint: maligned
 type nsIndex struct {
-	sync.RWMutex
-	insertMode index.InsertMode
-	state      nsIndexState
-	active     index.Block
+	sync.RWMutex         // NB: this mutex guards all fields until (and including) latestBlock
```

Comment:
Perhaps we should create a `nsIndexState` and use that to embed the `sync.RWMutex` there to represent this in code?
```go
i.state.Lock()
i.state.currState = / * Open/Closed */
i.state.latestBlock = /* ... */
i.state.Unlock()
```

Reply:
Done